feat: [MLT-48] Object crops on PDFs

Jim-Encord opened the pull request:
Feature, Tests, Docs not yet.

Currently adds dependency. Could technically become optional dependency

Jim-Encord committed:
fix: Fix for GCP. Test coverage on FastAPI

Jim-Encord committed:
feat: Object crops on PDFs

Jim-Encord commented on encord_agents/core/pdf.py:
```diff
@@ -0,0 +1,23 @@
+import pymupdf
+
+
+def extract_page(
```
Can't really be done by like other method as doc.close() apparently divorces pix so best to cast to file immediately

frederik-encord replied:
I think that we should fix this straight away. The `pumupdf.Pixmap` has a [`pil_image`](https://github.com/pymupdf/PyMuPDF/blob/cc114d71367c44df23b8fe7eb9889f6ade8d7f6c/src/__init__.py#L10253) method so we can make the optional dependencies `[mymupdf, PILLOW]` and then use that. Then, no need to store on disk.

Jim-Encord committed:
tests: Improve test coverage on object crops for videos

Jim-Encord commented:
On a high level, this would be improved if instead we didn't write the intermediary crops to a file. As is, I'm just emulating the behaviour for videos but this could be improved for sure. Ticket already made around: Serverlessness

frederik-encord commented on pyproject.toml:
```diff
@@ -56,7 +56,7 @@ typing-extensions = ">=4.8.0"
 numpy = ">=1.26.4"
 opencv-python = ">=4.1"
-
+PyMuPdf = ">1.25.0"
```
This is a big no no IMO. I agree with your comment that this should be made optional.
So you do `pip install "encord-agents[pdf]" if you need it.
`encord-agents` has to be light weight in terms of dependencies. Cause what happens when we need to support DICOM or some exotic geospatial data type.

Jim-Encord committed:
fix: Introduce pdf optional group.

Gives support for pdf package and provides method for interfacing with pdfs.
Makes this package optional and slightly modify CI to ensure testing With pdf support
